File: AOM/packages/aom-protocol-ts/src/artifact.ts
export type ArtifactContainerType =
  | "directory"
  | "mac_app_bundle"
  | "pe"
  | "mach_o"
  | "elf"
  | "zip"
  | "asar"
  | "apk"
  | "app_image"
  | "unknown";

export type WebRuntimeFamily =
  | "electron"
  | "cef"
  | "web_view2"
  | "nwjs"
  | "tauri"
  | "qt_web_engine"
  | "generic_web"
  | "unknown";

export interface RuntimeCandidate {
  runtime: WebRuntimeFamily;
  confidence: number;
}

export interface ArtifactDetectionEvidence {
  evidenceId: string;
  detector: string;
  kind: string;
  value: string;
  locator?: string;
}

export interface ArtifactInspection {
  inspectionId: string;
  inputLocator: string;
  containerType: ArtifactContainerType;
  architecture?: string;
  runtimeCandidates: RuntimeCandidate[];
  recommendedAdapter?: string;
  evidence: ArtifactDetectionEvidence[];
}

File: AOM/crates/aom-protocol-rs/src/artifact.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum ArtifactContainerType {
    Directory,
    MacAppBundle,
    Pe,
    MachO,
    Elf,
    Zip,
    Asar,
    Apk,
    AppImage,
    Unknown,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum WebRuntimeFamily {
    Electron,
    Cef,
    WebView2,
    Nwjs,
    Tauri,
    QtWebEngine,
    GenericWeb,
    Unknown,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct RuntimeCandidate {
    pub runtime: WebRuntimeFamily,
    pub confidence: f32,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct ArtifactDetectionEvidence {
    pub evidence_id: String,
    pub detector: String,
    pub kind: String,
    pub value: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub locator: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct ArtifactInspection {
    pub inspection_id: String,
    pub input_locator: String,
    pub container_type: ArtifactContainerType,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub architecture: Option<String>,
    #[serde(default)]
    pub runtime_candidates: Vec<RuntimeCandidate>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub recommended_adapter: Option<String>,
    #[serde(default)]
    pub evidence: Vec<ArtifactDetectionEvidence>,
}
